### ISSUE TITLE
`interesting_values`   ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

### ISSUE THREAD

**Issue by pabloazurduy** [Featuretools/featuretools#356]
### Bug/Feature Request Description

Hi 
I'm using the interesting_values feature with some columns and I had an error that i can't trace:

```
Traceback (most recent call last):

  File "<ipython-input-118-24dc590a6faf>", line 1, in <module>
    runfile('myscript.py')

  File "C:\Anaconda3\envs\mantto\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 704, in runfile
    execfile(filename, namespace)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "myscript.py", line 128, in <module>
    training_window="15d"

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\synthesis\dfs.py", line 184, in dfs
    features = dfs_object.build_features(verbose=verbose)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\synthesis\deep_feature_synthesis.py", line 218, in build_features
    all_features, max_depth=self.max_depth)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\synthesis\deep_feature_synthesis.py", line 340, in _run_dfs
    max_depth=new_max_depth)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\synthesis\deep_feature_synthesis.py", line 365, in _run_dfs
    all_features, entity, max_depth=max_depth)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\synthesis\deep_feature_synthesis.py", line 497, in _build_transform_features
    self._add_identity_features(all_features, entity)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\synthesis\deep_feature_synthesis.py", line 452, in _add_identity_features
    new_f = IdentityFeature(variable=v)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\primitives\primitive_base.py", line 432, in __init__
    self.variable = variable.entityset.metadata[entity_id][variable.id]

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\entityset\entityset.py", line 186, in metadata
    data_root=None)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\entityset\entityset.py", line 263, in from_metadata
    if v['interesting_values'] and len(v['interesting_values']):

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
any guidance in this issue ?, i don't have a replication error code yet, but I'm working on it 

-----
*Issues created here on Github are for bugs or feature requests. For usage questions and questions about errors, please ask on Stack Overflow with the [featuretools](https://stackoverflow.com/questions/tagged/featuretools) tag. Check the [documentation](https://docs.featuretools.com/#get-help) for further guidance on where to ask your question.*

**Comment by kmax12** [comment]
the issue here is that you are setting interesting values to be a numpy array or pandas series. we should handle this case. 

in the meantime, can you cast interesting values to a python list when you assign it?

**Comment by pabloazurduy** [comment]
perfect ! casting solved the issue.
...
```
# my_entity_df (pd.DataFrame)
category_vars=['my_column_a', 'my_column_b']
for var in category_vars:
    es["myentity"][var].interesting_values = my_entity_df [var].unique()
```

new code:
```
# my_entity_df (pd.DataFrame)
category_vars=['my_column_a', 'my_column_b']
for var in category_vars:
    es["myentity"][var].interesting_values = list(my_entity_df [var].unique())
```

**Comment by kmax12** [comment]
one other thing - what is the variable type of the variable? If the variable is Categorical or Ordinal, the cast should have happened for you.

**Comment by pabloazurduy** [comment]
Now I defined all vars as  `ft.variable_types.Categorical` (a few of them where `np.float`), I removed the `list()`  casting, but I got a different error:
```

  File "<ipython-input-135-24dc590a6faf>", line 1, in <module>
    runfile('C:/Users/4252882/github/mantto/feature-creation/pfrs_aircraft.py')

  File "C:\Anaconda3\envs\mantto\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 704, in runfile
    execfile(filename, namespace)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/4252882/github/mantto/feature-creation/pfrs_aircraft.py", line 135, in <module>
    training_window="15d"

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\synthesis\dfs.py", line 200, in dfs
    verbose=verbose)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\computational_backends\calculate_feature_matrix.py", line 258, in calculate_feature_matrix
    pass_columns=pass_columns)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\computational_backends\calculate_feature_matrix.py", line 520, in linear_calculate_chunks
    backend=backend)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\computational_backends\calculate_feature_matrix.py", line 342, in calculate_chunk
    training_window=window)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\computational_backends\utils.py", line 34, in wrapped
    r = method(*args, **kwargs)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\computational_backends\calculate_feature_matrix.py", line 316, in calc_results
    profile=profile)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\computational_backends\pandas_backend.py", line 196, in calculate_all_features
    result_frame = handler(group, input_frames)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\computational_backends\pandas_backend.py", line 440, in _calculate_agg_features
    to_merge = base_frame.groupby(base_frame[groupby_var], observed=True, sort=False).apply(wrap)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\groupby\groupby.py", line 930, in apply
    return self._python_apply_general(f)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\groupby\groupby.py", line 936, in _python_apply_general
    self.axis)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\groupby\groupby.py", line 2273, in apply
    res = f(group)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\computational_backends\pandas_backend.py", line 518, in wrap
    d[f.get_name()] = [func(*args)]

  File "C:\Anaconda3\envs\mantto\lib\site-packages\featuretools\primitives\aggregation_primitives.py", line 172, in pd_topn
    return np.array(x.value_counts()[:n].index)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\series.py", line 810, in __getitem__
    return self._get_with(key)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\series.py", line 815, in _get_with
    indexer = self.index._convert_slice_indexer(key, kind='getitem')

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\indexes\numeric.py", line 341, in _convert_slice_indexer
    return self.slice_indexer(key.start, key.stop, key.step, kind=kind)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\indexes\base.py", line 4107, in slice_indexer
    kind=kind)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\indexes\base.py", line 4314, in slice_locs
    end_slice = self.get_slice_bound(end, 'right', kind)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\indexes\base.py", line 4244, in get_slice_bound
    raise err

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\indexes\base.py", line 4238, in get_slice_bound
    slc = self._get_loc_only_exact_matches(label)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\indexes\base.py", line 4207, in _get_loc_only_exact_matches
    return self.get_loc(key)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\indexes\numeric.py", line 419, in get_loc
    tolerance=tolerance)

  File "C:\Anaconda3\envs\mantto\lib\site-packages\pandas\core\indexes\base.py", line 3080, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))

  File "pandas\_libs\index.pyx", line 140, in pandas._libs.index.IndexEngine.get_loc

  File "pandas\_libs\index.pyx", line 162, in pandas._libs.index.IndexEngine.get_loc

  File "pandas\_libs\hashtable_class_helper.pxi", line 379, in pandas._libs.hashtable.Float64HashTable.get_item

  File "pandas\_libs\hashtable_class_helper.pxi", line 385, in pandas._libs.hashtable.Float64HashTable.get_item

KeyError: 3.0
```
it's possible that that error is related with the `np.float` nature of the column ?

**Comment by kmax12** [comment]
this is an issue with the NMostCommon primitive. can you remove that for now?

**Comment by pabloazurduy** [comment]
Perfect, now it worked! :+1: 

BTW ¿Why it worked when the column is not defined as `ft.variable_types.Categorical`? 

Thanks

**Comment by kmax12** [comment]
the primitive only gets applied to categoricals values. if the column isn't marked at cateogrical it wouldn't get applied and this error during calculation doesn't occur.

**Comment by rwedge** [comment]
Looks like this was fixed in release v0.7.0
